channeldb/graph.go

```diff
@@ -3616,9 +3669,34 @@ func putChanEdgePolicy(edges, nodes *bbolt.Bucket, edge *ChannelEdgePolicy,
 		return err
 	}
 
+	updateEdgePolicyDisabledIndex(edges, edge.ChannelID,
+		edge.ChannelFlags&lnwire.ChanUpdateDirection > 0, edge.IsDisabled())
+
 	return edges.Put(edgeKey[:], b.Bytes()[:])
 }
 
+func updateEdgePolicyDisabledIndex(edges *bbolt.Bucket, chanID uint64,
+	initiator bool, disabled bool) error {
+
+	var disabledEdgeKey [8 + 1]byte
+	byteOrder.PutUint64(disabledEdgeKey[0:], chanID)
+	if initiator {
+		disabledEdgeKey[8] = 1
+	}
+
+	disabledEdgePolicyIndex, err := edges.CreateBucketIfNotExists(disabledEdgePolicyBucket)
+	if err != nil {
+		return err
+	}
+
+	if disabled {
+		err = disabledEdgePolicyIndex.Put(disabledEdgeKey[:], nil)
```

Comment:
we never want to pass `nil` as the value to bolt, instead we should pass `[]byte{}`. bolt represents buckets as keys with nil values, which can cause undefined behavior if values are also nil. it is accepted in order to maintain backwards compatibility, but is generally not advised

Reply:
I am aware of that now. fixed.